drivers/serial/serial_esp32_usb.c

```diff
@@ -264,27 +232,14 @@ static const DRAM_ATTR struct uart_driver_api serial_esp32_usb_api = {
 #endif /* CONFIG_UART_INTERRUPT_DRIVEN */
 };
 
-#define ESP32_UART_INIT(idx)                                                                       \
-	static const DRAM_ATTR struct serial_esp32_usb_config serial_esp32_usb_cfg_port_##idx = {  \
-		.uart_num = DT_INST_PROP(idx, peripheral),                                         \
-		.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(idx)),                              \
-		.clock_subsys = (clock_control_subsys_t)DT_INST_CLOCKS_CELL(idx, offset),          \
-		.irq_source = DT_INST_IRQN(idx)                                                    \
-	};                                                                                         \
-												   \
-	static struct serial_esp32_usb_data serial_esp32_usb_data_##idx = {                        \
-		.serial_config = { .baudrate = DT_INST_PROP(idx, current_speed),                   \
-				   .parity = UART_CFG_PARITY_NONE,                                 \
-				   .stop_bits = UART_CFG_STOP_BITS_1,                              \
-				   .data_bits = UART_CFG_DATA_BITS_8,                              \
-				   .flow_ctrl =                                                    \
-					   COND_CODE_1(DT_NODE_HAS_PROP(idx, hw_flow_control),     \
-						       (UART_CFG_FLOW_CTRL_RTS_CTS),               \
-						       (UART_CFG_FLOW_CTRL_NONE)) },               \
-	};                                                                                         \
-												   \
-	DEVICE_DT_INST_DEFINE(idx, &serial_esp32_usb_init, NULL, &serial_esp32_usb_data_##idx,     \
-			      &serial_esp32_usb_cfg_port_##idx, PRE_KERNEL_1,                      \
-			      CONFIG_SERIAL_INIT_PRIORITY, &serial_esp32_usb_api);
-
-DT_INST_FOREACH_STATUS_OKAY(ESP32_UART_INIT);
+static const DRAM_ATTR struct serial_esp32_usb_config serial_esp32_usb_cfg = {
+	.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(0)),
+	.clock_subsys = (clock_control_subsys_t)DT_INST_CLOCKS_CELL(0, offset),
+	.irq_source = DT_INST_IRQN(0)
+};
+
+static struct serial_esp32_usb_data serial_esp32_usb_data_0;
+
+DEVICE_DT_INST_DEFINE(0, &serial_esp32_usb_init, NULL, &serial_esp32_usb_data_0,
```

Comment:
Is there only ever one?  if not, looks like a step back for this not to use `DT_INST_FOREACH_STATUS_OKAY`

Reply:
Yes, it's a very special interface with fixed pins and a built-in USB CDC ACM + JTAG. It cannot be used as a "normal" USB peripheral and is mainly intended for flashing and debugging. Don't think it would make sense to ever have two of these in one MCU. @sylvioalves please correct me if I'm wrong.

Reply:
Correct!